WalletWasabi/Bases/PeriodicRunner.cs

```diff
@@ -56,17 +57,21 @@ public void TriggerRound()
 
 		public void Start()
 		{
-			ForeverTask = ForeverMethodAsync();
+			ForeverTask = ForeverMethodAsync(x => false);
 		}
 
-		private async Task ForeverMethodAsync()
+		protected async Task ForeverMethodAsync(Func<T, bool> finishIf)
 		{
 			while (!Stop.IsCancellationRequested)
 			{
 				try
 				{
 					Status = await ActionAsync(Stop.Token).ConfigureAwait(false);
```

Comment:
I would remove `Func<T, bool> finishIf` and add an out variable here: 
`ActionAsync(Stop.Token, out bool isFinished);`

Reply:
discussed